carma_wm/src/WMListenerWorker.cpp

```diff
@@ -52,8 +52,21 @@ void WMListenerWorker::mapCallback(const autoware_lanelet2_msgs::MapBinConstPtr&
     map_callback_();
   }
 }
+
+bool WMListenerWorker::checkIfReRoutingNeeded()
```

Comment:
This method works as a way for the route node to know if it needs to update the route, but there is still no mechanism to allow the route node specifically to update the map while other nodes do not. You need something the route node can set in the wm listener to handle that.